src/test/regress/expected/multi_colocation_utils.out

```diff
@@ -343,3 +343,227 @@ SELECT find_shard_interval_index(1300016);
                          0
 (1 row)
 
+-- check external colocation API
+SET citus.shard_count = 2;
+CREATE TABLE table1_groupA ( id int );
+SELECT create_distributed_table('table1_groupA', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE table2_groupA ( id int );
+SELECT create_distributed_table('table2_groupA', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- change shard replication factor
+SET citus.shard_replication_factor = 1;
+CREATE TABLE table1_groupB ( id int );
+SELECT create_distributed_table('table1_groupB', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE table2_groupB ( id int );
+SELECT create_distributed_table('table2_groupB', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- revert back to default shard replication factor
+SET citus.shard_replication_factor to DEFAULT;
+-- change partition column type
+CREATE TABLE table1_groupC ( id text );
+SELECT create_distributed_table('table1_groupC', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE table2_groupC ( id text );
+SELECT create_distributed_table('table2_groupC', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- change shard count
+SET citus.shard_count = 4;
+CREATE TABLE table1_groupD ( id int );
+SELECT create_distributed_table('table1_groupD', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE table2_groupD ( id int );
+SELECT create_distributed_table('table2_groupD', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- try other distribution methods
+CREATE TABLE table_append ( id int );
+SELECT create_distributed_table('table_append', 'id', 'append');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE table_range ( id int );
+SELECT create_distributed_table('table_range', 'id', 'range');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- test foreign table creation
+CREATE FOREIGN TABLE table3_groupD ( id int ) SERVER fake_fdw_server;
+SELECT create_distributed_table('table3_groupD', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- check metadata
+SELECT * FROM pg_dist_colocation 
+    WHERE colocationid >= 1 AND colocationid < 1000 
+    ORDER BY colocationid;
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+            1 |          2 |                 2 |                     23
+            2 |          2 |                 1 |                     23
+            3 |          2 |                 2 |                     25
```

Comment:
Haven't we used `DEFAULT` for group C? Why we see `shardcount` as 2 (i.e., isn't default 32)?